### PR TITLE
Side exercises for twelve days 1 of 3

### PR DESCRIPTION
**Pull request by glennj**
| exercise | canonical data | difficulty |
| --- | --- | :-: |
| [bowling](https://github.com/glennj/tcl/blob/side-exercises-for-twelve-days-1-of-3/exercises/bowling/README.md) | [canonical-data.json](https://github.com/exercism/problem-specifications/blob/master/exercises/bowling/canonical-data.json) | 5 |
| [complex-numbers](https://github.com/glennj/tcl/blob/side-exercises-for-twelve-days-1-of-3/exercises/complex-numbers/README.md) | [canonical-data.json](https://github.com/exercism/problem-specifications/blob/master/exercises/complex-numbers/canonical-data.json) | 3 |
| [diamond](https://github.com/glennj/tcl/blob/side-exercises-for-twelve-days-1-of-3/exercises/diamond/README.md) | [canonical-data.json](https://github.com/exercism/problem-specifications/blob/master/exercises/diamond/canonical-data.json) | 3 |
| [grep](https://github.com/glennj/tcl/blob/side-exercises-for-twelve-days-1-of-3/exercises/grep/README.md) | [canonical-data.json](https://github.com/exercism/problem-specifications/blob/master/exercises/grep/canonical-data.json) | 6 |
| [yacht](https://github.com/glennj/tcl/blob/side-exercises-for-twelve-days-1-of-3/exercises/yacht/README.md) | [canonical-data.json](https://github.com/exercism/problem-specifications/blob/master/exercises/yacht/canonical-data.json) | 4 |

Part 1 of 3.

yacht is a side exercise for resistor-color

### CODE REVIEW

**Comment by glennj** [comment]
@sshine, are you swamped, or can you give this PR some attention?

**Comment by sshine** [comment]
I'll get a round tuit this weekend!

**Comment by sshine** [comment]
How many exercises remain in your backlog at this point?

File: exercises/complex-numbers/complex-numbers.test [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+customMatch pairOfFloats pairFloatMatch
+proc pairFloatMatch {expected actual {precision 8}} {
+    lassign $expected r1 i1
+    lassign $actual   r2 i2
+    return [expr {
+        [format "%.${precision}f" $r1] == [format "%.${precision}f" $r2] &&
+        [format "%.${precision}f" $i1] == [format "%.${precision}f" $i2]
+    }]
+}
```
[the remark this answers is not in the document]
Comment (glennj) [reply exercises/complex-numbers/complex-numbers.test:40]: Not specifically, but I can add a `floatMatch` matcher, and build this complex matcher out of it.

**Pull request merged**